### PR TITLE
Update to smallrye-jwt 1.1.4 and remove CDI producers

### PR DESCRIPTION
**Pull request by sberyozkin**
The CDI producers code from Quarkus and Thorntail smallrye-jwt extensions has been moved to smallrye-jwt, see https://github.com/smallrye/smallrye-jwt/pull/42

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -50,7 +50,7 @@
         <smallrye-open-api.version>1.1.1</smallrye-open-api.version>
         <smallrye-opentracing.version>1.3.0</smallrye-opentracing.version>
         <smallrye-fault-tolerance.version>2.0.4</smallrye-fault-tolerance.version>
-        <smallrye-jwt.version>1.1.3</smallrye-jwt.version>
+        <smallrye-jwt.version>1.1.4-SNAPSHOT</smallrye-jwt.version>
```
Comment (gsmet) [inline bom/pom.xml]: Hmmm, is it a work in progress? Because we don't want a SNAPSHOT dependency there?

Other than that, I'll let @starksm64 review it.

**Review by gsmet — DISMISSED** [review]
We sure do not want SNAPSHOT dependencies.

File: bom/pom.xml (hunk 1 above)
Comment (gsmet) [reply bom/pom.xml]: Ah, OK, I was too fast :).

**Review by starksm64 — COMMENTED** [review]
I have verified that this works with the snapshot, so once the smallrye-jwt is ready for release, this an be updated to refer to the release and merged.

File: bom/pom.xml (hunk 1 above)
Comment (sberyozkin) [reply bom/pom.xml]: @gsmet np, yes, it is a work in progress, I added the label (coding in progress).

Comment (sberyozkin) [reply bom/pom.xml]: I see, will add the [DO NOT MERGE] in the future

**Review by starksm64 — APPROVED** [review]

**Review by starksm64 — APPROVED** [review]

**Pull request merged**